src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { launchBrowser } from '../../common/browser-factory';
+import { fastPassSelectors } from '../../common/element-identifiers/common-selectors';
+import { popupPageElementIdentifiers } from '../../common/element-identifiers/popup-page-element-identifiers';
+import { Page } from '../../common/page-controllers/page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { Browser } from './../../common/browser';
+import { DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS } from './../../common/timeouts';
+
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let fastPassPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            fastPassPage = await gotoFastPass(browser, targetPage.tabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            const toggleAriaLabel = 'Tab stops';
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            await targetPage.bringToFront();
+
+            // press tabs 3 times. todo: DRY this code
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function gotoFastPass(browserLocal: Browser, targetTabId: number): Promise<Page> {
+            const popupPage = await browserLocal.newPopupPage(targetPage);
+            let fastPass: Page;
+            await Promise.all([
+                browserLocal.waitForDetailsViewPage(targetPage).then(page => (fastPass = page)),
+                popupPage.clickSelector(popupPageElementIdentifiers.fastPass),
+            ]);
+
+            return fastPass;
+        }
+
+        async function goToTabStopTest(): Promise<void> {
+            await fastPassPage.clickSelector(fastPassSelectors.tabstopNavButtonSelector);
+            // await fastPassPage.disableAnimations();
+            await verifyTabStopPageLoaded();
```

Comment:
`verifyTabStopPageLoaded` is a thin wrapper around an specific `waitForSelector` call... the abstraction level here seems off between lines 57 and 59 (57 has more details than 59) I'm wondering if we actually need to have `verifyTabStopPageLoaded` in the first place

Reply:
We do need it, just to wait for the radio button to load and do its animation stuff. Without a couple of waitForSelectors, I found that the click event was very flaky!